### PR TITLE
Config overhaul part 2 electric boogaloo

### PR DESCRIPTION
**Pull request by BoogieMonster1O1**
- Config overhaul. Now has two categories and classes for common and client configs. Everything is still in the same file, though.  
- Added an option to disable the enchantment glint on potions. Disabled by default. 
- Added an option to allow colored sheep to naturally spawn. Enabled by default. 
- Added an option to optimized bamboo rendering. Enabled by default.

### CODE REVIEW

**Comment by redcreeper14385** [comment]
@YTG1234 can you review this as well as me?

**Review by YTG1234 — APPROVED** [review]
Looks fine to me

File: src/main/java/mounderfod/moundertweaks/mixin/client/OptionScreenMixin.java [hunk 1]
```diff
@@ -34,8 +34,8 @@ private OptionScreenMixin(Text title) {
     @Inject(at = @At("RETURN"), method = "init")
     public void drawConfigScreen(CallbackInfo ci) {
         // Only load if modmenu is not present
-        if (!FabricLoader.getInstance().isModLoaded("modmenu")) {
-            this.addButton(new ButtonWidget(this.width / 2 - 100, this.height / 6 + 192, 200, 20, new TranslatableText("gui.moundertweaks.config"), (buttonWidget) -> {
+        if (!FabricLoader.getInstance().isModLoaded("modmenu") && this.client != null) {
+            this.addButton(new ButtonWidget(this.width / 2 - 100, this.height / 6 + 192, 200, 20, new TranslatableText("text.autoconfig.moundertweaks.title"), (buttonWidget) -> {
```
Comment (redcreeper14385) [inline src/main/java/mounderfod/moundertweaks/mixin/client/OptionScreenMixin.java:38]: Where is this button created?

**Review by redcreeper14385 — APPROVED** [review]
Looks good to me, merging now.
Be aware that this won't be pushed to the end user for quite some time, I'll be adding more features and then I'll build JARs, etc.

**Pull request merged**